llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -512,9 +513,19 @@ void VPInstruction::execute(VPTransformState &State) {
     if (!hasResult())
       continue;
     assert(GeneratedValue && "generateInstruction must produce a value");
-    State.set(this, GeneratedValue, Part);
+    if (GeneratedValue->getType()->isVectorTy()) {
+      State.set(this, GeneratedValue, Part);
```

Comment:
nit: can fold into a single
```suggestion
    bool isVector = GeneratedValue->getType()->isVectorTy();
    State.set(this, GeneratedValue, Part, !isVector);
    assert((isVector || getOpcode() == VPInstruction::ComputeReductionResult ||
                  State.VF.isScalar() || vputils::onlyFirstLaneUsed(this)) &&
             "scalar value but not only first lane used");
```

Reply:
Adjusted, thanks!